app/models/chore.rb

```diff
@@ -1,7 +1,24 @@
 class Chore < ApplicationRecord
+  enum interval_type: { calendar_days: 0, business_days: 1 }
+
   validates :name, presence: true
   validates :interval_days, numericality: { only_integer: true, greater_than: 0 }
 
+  def reset_cycle_date
+    self.last_done = Date.today
+    self.cycle_reset = Time.now
+    case
```

Comment:
What's the purpose of moving the `case` statement into its own function? While I will certainly do so, and have ideas for where to divide responsibility, I'm not sure what benefit you're aiming for, and thus not sure if I'll be achieving it.

Regarding returning `true` and `false`, I did that essentially as a debug step. Upon reflection, I should probably pull that out.